### PR TITLE
Reduce complexity in ConditionalAssignment and IndentationWidth

### PR DESCRIPTION
**Pull request by pocke**
-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/indentation_width.rb [hunk 1]
```diff
@@ -22,19 +22,18 @@ class IndentationWidth < Cop
...
-            _, _, body = *rescue_node
-            check_indentation(rescue_node.loc.keyword, body)
-          end
+          _begin_node, *_rescue_nodes, else_node = *node
           check_indentation(node.loc.else, else_node)
         end
 
         def on_ensure(node)
-          _body, ensure_body = *node
+          ensure_body = node.children.last
```
Comment (bbatsov) [inline lib/rubocop/cop/style/indentation_width.rb]: I think the original code was clearer and a bit faster.

File: lib/rubocop/cop/style/conditional_assignment.rb [hunk 2]
```diff
@@ -31,16 +31,14 @@ def tail(branch)
...
           when :send
             lhs_for_send(node)
           when :op_asgn
             "#{node.children[0].source} #{node.children[1]}= "
-          when :and_asgn
-            "#{node.children[0].source} &&= "
-          when :or_asgn
-            "#{node.children[0].source} ||= "
+          when :and_asgn, :or_asgn
+            "#{node.children[0].source} #{node.loc.operator.source} "
```
Comment (Drenmi) [inline lib/rubocop/cop/style/conditional_assignment.rb]: With the new node extensions, you could now do:

```
"#{node.lhs.source} #{node.loc.operator.source} "
```

😊

**Comment by pocke** [comment]
> I think the original code was clearer and a bit faster.


I've fixed for the suggestion in f76cc6827a66ae85185b33e3511c8194c21b077c . However, I can't be sure this change is expected by you. This change is as expected? Or Should I revert the entire `indentation_width.rb` ?

**Comment by pocke** [comment]
https://github.com/bbatsov/rubocop/commit/f76cc6827a66ae85185b33e3511c8194c21b077c#commitcomment-20871750

I've modify the `on_ensure` method to avoid unnamed variable by the above comment way in d449907.

File: lib/rubocop/ast/node/ensure_node.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+module RuboCop
+  module AST
+    # A node extension for `ensure` nodes. This will be used in place of a plain
+    # node when the builder constructs the AST, making its methods available
+    # to all `ensure` nodes within RuboCop.
+    class EnsureNode < Node
+      # Returns the body of the `ensure` clause.
+      #
+      # @return [Node, nil] The body of the `for` loop.
```
Comment (Drenmi) [inline lib/rubocop/ast/node/ensure_node.rb]: Forgot to change the YARD here. 🙂

File: lib/rubocop/ast/node/resbody_node.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+module RuboCop
+  module AST
+    # A node extension for `resbody` nodes. This will be used in place of a
+    # plain node when the builder constructs the AST, making its methods
+    # available to all `resbody` nodes within RuboCop.
+    class ResbodyNode < Node
+      # Returns the body of the `rescue` clause.
+      #
+      # @return [Node, nil] The body of the `for` loop.
```
Comment (Drenmi) [inline lib/rubocop/ast/node/resbody_node.rb]: Forgot to change the YARD here. 🙂

**Comment by Drenmi** [comment]
@pocke: Would love if you could add unit tests for the node extensions. You can use the existing ones for reference. ❤️

**Pull request merged**